tests/unit/test_git_url_watcher.py

```diff
@@ -837,3 +866,224 @@ async def test_git_url_watcher_rolls_back_if_tag_on_remote_vanishes_tag_sync(
         )
 
     await git_watcher.cleanup()
+
+
+async def test_git_url_watcher_cannot_init_when_no_commit_on_remote(
+    event_loop: AbstractEventLoop,
+    git_config_tags: dict[str, Any],
+):
+    git_watcher = git_url_watcher.GitUrlWatcher(git_config_tags)
+    # Cannot init if no file present on remote
+    with pytest.raises(ConfigurationError):
+        init_result = await git_watcher.init()
+
+
+async def test_git_url_watcher_tag_sync_with_multiple_tags_per_commit(
+    event_loop: AbstractEventLoop,
+    git_config_two_repos_synced_same_tag_regex: dict[str, Any],
+):
+    assert git_config_two_repos_synced_same_tag_regex["main"]["synced_via_tags"]
+    git_watcher = git_url_watcher.GitUrlWatcher(
+        git_config_two_repos_synced_same_tag_regex
+    )
+
+    # add a file, commit, and tag
+    VALID_TAG: Literal["staging_z1stvalid"] = "staging_m1stvalid"
+    TESTFILE_NAME: Literal["testfile.csv"] = "testfile.csv"
+    sleep_1_sec_to_make_commit_timestamp_unique()
+    _helper_list_watched_repos = [
+        git_config_two_repos_synced_same_tag_regex["main"]["watched_git_repositories"][
+            i
+        ]
+        for i in range(
+            len(
+                git_config_two_repos_synced_same_tag_regex["main"][
+                    "watched_git_repositories"
+                ]
+            )
+        )
+    ]
```

Comment:
for a next time: I guess this and also the part in lines 906-910 could be made into fixtures for better readability.
a fixture called ```watched_repositories```
and another ```add_files_to_repo -> Callback[int, str, None]``` (you would call by doing add_files_to_repo(3, repo))

Reply:
next time, thx ;)